### ISSUE TITLE
An issue about setAttestationChallenge() in "android-key" attestation statement

### ISSUE THREAD

**Issue by caileb2017** [w3c/webauthn#456]
I'm testing about "android-key" attestation statement of FIDO2.0 and I got an issue.
The test code that I tried it is like the code below.
```
String challenge = "f9084f24ac8c33 ... 3c6ef635cdd75746516"; // **this data size is 103 bytes**
KeyGenParameterSpec.Builder builder = new KeyGenParameterSpec.Builder(name, KeyProperties.PURPOSE_SIGN )
   .setAttestationChallenge(challenge.getBytes())
   .setUserAuthenticationRequired(true)
   .setDigests(KeyProperties.DIGEST_SHA256, KeyProperties.DIGEST_SHA512)
   .setAlgorithmParameterSpec(new ECGenParameterSpec("secp256r1"));
mKeyGenerator.initialize(builder.build());
mKeyGenerator.generateKeyPair();
```

and it occures an error below.
```
java.security.ProviderException: Failed to generate attestation certificate chain
    at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.getAttestationChain(AndroidKeyStoreKeyPairGeneratorSpi.java:631)
    at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.createCertificateChain(AndroidKeyStoreKeyPairGeneratorSpi.java:519)
    at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.generateKeyPair(AndroidKeyStoreKeyPairGeneratorSpi.java:502)
    at java.security.KeyPairGenerator$Delegate.generateKeyPair(KeyPairGenerator.java:699)
    ...
Caused by: android.security.KeyStoreException: Invalid input length
    at android.security.KeyStore.getKeyStoreException(KeyStore.java:1097)
    at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.getAttestationChain(AndroidKeyStoreKeyPairGeneratorSpi.java:632)
    at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.createCertificateChain(AndroidKeyStoreKeyPairGeneratorSpi.java:519) 
    at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.generateKeyPair(AndroidKeyStoreKeyPairGeneratorSpi.java:502) 
    at java.security.KeyPairGenerator$Delegate.generateKeyPair(KeyPairGenerator.java:699) 
    ...
```

I used "setAttestationChallenge()" function as mentioned FIDO Specification on W3C.
I think that the function has a limit size and it is smaller than the size of our challege value(attToBeSigned = authenticatorData + clientHash).
how do you think about this case?

**Comment by AngeloKai** [comment]
Assigning this to @leshi because this relates to Android attestation. Moving this to CR milestone because it appears to be a bug in the spec.

**Comment by equalsJeffH** [comment]
@AngeloKai, by "the spec" do you mean the (submitted) "FIDO Specification on W3C" (site) ?  I am guessing that is what @caileb2017 is referring to.  

although, there is no "setAttestationChallenge()" in the submitted FIDO 2 specs. 

I did find it in [fido-uaf-reg-v1.1-rd-20161005.html](https://fidoalliance.org/specs/fido-uaf-v1.1-rd-20161005/fido-uaf-reg-v1.1-rd-20161005.html) which is now superseded. Plus it occurs once in an example, and thus non-normative. 

We should just decline this issue.

**Comment by AngeloKai** [comment]
@equalsJeffH Interesting. I guess I don't have enough context here with regarding to Android attestation and UAF. Perhaps @leshi has more context.

**Comment by christiaanbrand** [comment]
Fixed by PR #546
